sycl/ReleaseNotes.md

```diff
@@ -1,3 +1,172 @@
+# Nov'23 release notes
+Release notes for commit range f4e0d3177338..f4ed132f243a
+
+## New Features
+### SYCL Compiler
+- Added `-fpreview-breaking-changes` option [08febcfcb64a] [99b97eef4278]
```

Comment:
@steffenlarsen, @stdale-intel, should we elaborate here a bit more about the option purpose? In any case we can probably add a reference to a section where we describe changes we did under that flag